### PR TITLE
Port to topkg and clean up some warnings

### PR DESCRIPTION
**Pull request by avsm**

### CODE REVIEW

File: _tags [hunk 1]
```diff
@@ -1,44 +1,6 @@
...
-<lib_test/*.ml{,i,y}>: pkg_lwt.unix
-<lib_test/*.ml{,i,y}>: pkg_mirage-clock-unix
-<lib_test/*.ml{,i,y}>: pkg_mirage-profile
-<lib_test/*.ml{,i,y}>: pkg_mirage-types
-<lib_test/*.ml{,i,y}>: pkg_ptime
-<lib_test/*.ml{,i,y}>: use_mirage_logs
-# OASIS_STOP
 true: warn(A), strict_sequence
+true : bin_annot, safe_string, package(bytes)
+true : package(logs), package(mirage-types), package(lwt), package(mirage-profile), package(ptime)
```
Comment (samoht) [inline _tags]: you can use the shorter:

```
package(logs mirage-types lwt mirage-profile ptime)
```

File: .ocp-indent [hunk 2]
```diff
@@ -0,0 +1 @@
+strict_with=always,match_clause=4,strict_else=never
```
Comment (samoht) [inline .ocp-indent]: can we keep the default indentation?

File: doc/dev.odocl [hunk 3]
```diff
@@ -0,0 +1 @@
+Mirage_logs
```
Comment (samoht) [inline doc/dev.odocl:1]: as this is the same as `api.odocl` I am not sure you need this file.

File: test/test.ml [hunk 4]
```diff
@@ -68,8 +70,8 @@ let test_ring () =
     Logs.(set_level (Some Debug));
     Logs_reporter.(create ~ch:w ~ring_size:2 ~console_threshold clock |> run) @@ fun () ->
     test_console r;
-    Lwt.fail (Failure "Oops")
-  end with Failure "Oops" ->
+    Lwt.fail (Test_failure "Oops")
```
Comment (samoht) [inline test/test.ml]: maybe remove the string argument completely? e.g. `exception Test_failure` and `Lwt.fail Test_failure`

**Review by samoht — APPROVED** [review]

File: _tags (hunk 1 above)
Comment (avsm) [reply _tags]: fixed in 655003d6bdf99501b925a3560dbdc1f4ecb44b6c

**Comment by avsm** [comment]
No need for a release until mirage3 tags are cut. /cc @yomimono

**Pull request merged**